zkvm/src/types.rs

```diff
@@ -310,6 +323,25 @@ impl Expression {
     }
 }
 
+impl Neg for Expression {
+    type Output = Expression;
+
+    fn neg(self) -> Expression {
+        match self{
+            Expression::Constant(a) => {
+                Expression::Constant(-a)
+            }, 
+            Expression::Terms(terms,assignment)=> {
+
+                for (_, n) in terms.iter_mut() {
+                    -*n;
```

Comment:
you sure it changes n in place? Maybe `*n = -n`.?

Reply:
I impl Neg for ScalarWitness so I think it should. *n = -n is much clearer though, let me change